clouddriver-gae/src/main/groovy/com/netflix/spinnaker/clouddriver/gae/security/GaeNamedAccountCredentials.groovy

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.gae.security
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.google.api.services.appengine.v1.Appengine
+import com.netflix.spinnaker.clouddriver.gae.GaeCloudProvider
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials
+import groovy.transform.TupleConstructor
+
+@TupleConstructor
+class GaeNamedAccountCredentials implements AccountCredentials<GaeCredentials> {
+
+  final String name
+  final String environment
+  final String accountType
+  final String project
+  final String cloudProvider = GaeCloudProvider.GAE
+  final List<String> requiredGroupMembership
+
+  @JsonIgnore
+  final GaeCredentials credentials
+  final String applicationName
+  final Appengine appengine
+
+  static class Builder {
+    String name
+    String environment
+    String accountType
+    String project
+    List<String> requiredGroupMembership
+    GaeCredentials credentials
+
+    String jsonKey
+    String applicationName
+    Appengine appengine
+
+    Builder name(String name) {
+      this.name = name
+      return this
+    }
+
+    Builder environment(String environment) {
+      this.environment = environment
+      return this
+    }
+
+    Builder accountType(String accountType) {
+      this.accountType = accountType
+      return this
+    }
+
+    Builder project(String project) {
+      this.project = project
+      return this
+    }
+
+    Builder requiredGroupMembership(List<String> requiredGroupMembership) {
+      this.requiredGroupMembership = requiredGroupMembership
+      return this
+    }
+
+    Builder jsonKey(String jsonKey) {
+      this.jsonKey = jsonKey
+      return this
+    }
+
+    Builder applicationName(String applicationName) {
+      this.applicationName = applicationName
+      return this
+    }
+
+    GaeNamedAccountCredentials build() {
+      credentials = jsonKey ?
```

Comment:
If `jsonKey` is supplied, `GaeJsonCredentials(..)` will return something of type `GoogleCredential`, although the local `credentials` is of type `GaeCredentials`. Seems like this won't work?